### ISSUE TITLE
Unable to run tests on target

### ISSUE THREAD

**Issue by dfreiberger** [philippleidig/TcUnit-VsTestAdapter#1]
@philippleidig I've been playing with this a bit more, however I'm not able to run tests on a target (Beckhoff C6017 with Windows 10, or a VM running TcBSD). I can open the test project and deploy through TwinCAT, however when running with `vstest.console.exe` I get an error such as `Download CurrentConfig.xml to target failed with reason: Download CurrentConfig.xml to target failed`

As an example, here is code that I am running https://github.com/dfreiberger/TcUnit-VsTestAdapter/blob/ff528a799d30f3638f9001c58a5eb2afa0ead553/tests/TcUnit.TestAdapter.Tests/TestRunnerTests.cs#L60C1-L64

Are there any pre-requisites to running this on an IPC that I might be missing? I've not spent too much time troubleshooting yet.

I also had issues with test discovery, it seems like it will not work to call `vstest.console.exe /Settings:TcUnit.runsettings /ListTests *.tsproj` without this addition https://github.com/dfreiberger/TcUnit-VsTestAdapter/blob/ff528a799d30f3638f9001c58a5eb2afa0ead553/src/TcUnit.TestAdapter/Discovery/TestDiscoverer.cs#L54C10-L55, because otherwise the filter always returns false due to `successfullyGotFilter` being set to false by `GetTestCaseFilterExpressionFromDiscoveryContext`

```
  public bool MatchTestCase(TestCase testCase)
  {
      if (!successfullyGotFilter)
      {
          // Had an error while getting filter, match no testcase to ensure discovered test list is empty
          return false;
      }
```

**Comment by dfreiberger** [comment]
I was eventually able to get this working after some modifications to the code. The most reliable way to get this running is to make sure that the boot folder is already empty, sometimes the `CleanUpBootDirectory` method was failing for me. 

![image](https://github.com/philippleidig/TcUnit-VsTestAdapter/assets/3814449/6dfbf8c4-f215-4586-83c2-8be6c7d43b8f)

**Comment by philippleidig** [comment]
@dfreiberger, thanks for your great work! That looks amazing! 🤯
I'm currently on travel but will be looking at your progress during next week.

**Comment by dfreiberger** [comment]
@philippleidig thanks! I did have a question for you - was there a reason you opted to not use the Automation Interface APIs for this? I can see some benefits in being able to run this on a system without TwinCAT XAE installed, but at least in our use case we would not save the _Boot folder in git, so XAE needs to be installed on the build / test server anyway. Using Automation Interface might simplify some of the code, especially work arounds needed to figure out the target architecture and the like.

**Comment by dfreiberger** [comment]
Closing this issue since it is no longer a problem. @philippleidig are Github Issues the best way to communicate about development on this repo? Are you on slack? You can join all of us over at https://sase.space/!

**Comment by philippleidig** [comment]
@dfreiberger, the original idea with the test adapter was to execute automated unit tests in TwinCAT in less than 30 seconds, no matter where on which system or in a virtual machine / CI pipeline.

The main reason for not using the TwinCAT Automation Interface was speed and reliability.
Currently a pipeline run on Azure DevOps takes about 5-6 minutes with a good chance of timeouts.

In addition, the TwinCAT Automation Interface requires a full installation of the TcXaeShell, which in turn requires more performance from the pipeline agents.

With the release of the TwinCAT User Mode Runtime, even faster execution of unit tests might be possible. However, this still needs to be investigated in detail.

Basically, however, I absolutely understand your concerns about building a project or creating a library without the TwinCAT Automation Interface.
To be honest, I hope for future changes and possibilities with, for example, a build via a CLI 😊

**Comment by philippleidig** [comment]
@dfreiberger I have now had a good look at your work and changes and could absolutely imagine adopting them. I really like the approach with the TMC file.
Would you be interested in creating a PR or working together on the implementation?

At the moment I don't have enough time to deal with the topic in depth. I'm hoping for a quiet Christmas time 😊

I have tried to join sase-space.slack.com. But so far without success. Do I need a specific e-mail address?

**Comment by dfreiberger** [comment]
@philippleidig I sent you an invite link just now on LinkedIn, can you please try that? If it doesn't work I can message the slack owner to take a look.

**Comment by dfreiberger** [comment]
Regarding collaboration, I am certainly interested, although I also have the problem of not having enough spare time to work on this. I will create a PR soon.
